
/EkfCalNode:
    ros__parameters:
        debug_log_level: 2
        data_logging_on: true
        body_data_rate: 100.0
        sim_params:
            seed: 0.0
            use_seed: true
            max_time: 10.0
            no_errors: false
            number_of_runs: 10
            run_number: 0
            truth_type: "cyclic"
            stationary_time: 5.0
            pos_frequency: [0.3, 0.5, 0.7]
            ang_frequency: [0.0, 0.0, 0.0]
            pos_offset: [0.0, 0.0, 0.0]
            ang_offset: [0.0, 0.0, 0.0]
            ang_amplitude: 0.1
            pos_amplitude: 0.1

        filter_params:
            process_noise: [
                1.0e-3, 1.0e-3, 1.0e-3, # Position
                1.0e-4, 1.0e-4, 1.0e-4, # Velocity
                1.0e-5, 1.0e-5, 1.0e-5, # Acceleration
                1.0e-3, 1.0e-3, 1.0e-3, # Angular Position
                1.0e-4, 1.0e-4, 1.0e-4, # Angular Velocity
                1.0e-5, 1.0e-5, 1.0e-5, # Angular Acceleration
            ]

        imu_list:
            - "vn300"
            - "vn100"

        camera_list:
            - "ace"

        tracker_list:
            - "orb"

        fiducial_list:

        gps_list:
            - "gps_1"
            - "gps_2"

        imu:
            vn300:
                use_for_prediction: false
                is_extrinsic: false
                is_intrinsic: false
                rate: 400.0
                topic: "/vn300/imu_uncompensated"
                variance: [0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01]
                pos_i_in_b: [0.0, 0.0, 0.0]
                ang_i_to_b: [1.0, 0.0, 0.0, 0.0]
                acc_bias: [0.0, 0.0, 0.0]
                omg_bias: [0.0, 0.0, 0.0]
                pos_stability: 1.0e-2
                ang_stability: 1.0e-2
                acc_bias_stability: 1.0e-3
                omg_bias_stability: 1.0e-3
                sim_params:
                    time_bias_error: 1.0e-3
                    time_skew_error: 1.0e-9
                    time_error: 1.0e-6
                    pos_error: [0.0, 0.0, 0.0]
                    ang_error: [0.0, 0.0, 0.0]
                    acc_error: [1.0e-3, 1.0e-3, 1.0e-3]
                    omg_error: [1.0e-2, 1.0e-2, 1.0e-2]
                    acc_bias_error: [1.0e-1, 1.0e-1, 1.0e-1]
                    omg_bias_error: [1.0e-1, 1.0e-1, 1.0e-1]
            vn100:
                use_for_prediction: false
                is_extrinsic: true
                is_intrinsic: false
                rate: 100.0
                topic: "/vn100/imu_uncompensated"
                variance: [0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01]
                pos_i_in_b: [0.0, 0.0, 0.0]
                ang_i_to_b: [1.0, 0.0, 0.0, 0.0]
                acc_bias: [0.0, 0.0, 0.0]
                omg_bias: [0.0, 0.0, 0.0]
                pos_stability: 1.0e-2
                ang_stability: 1.0e-2
                acc_bias_stability: 1.0e-3
                omg_bias_stability: 1.0e-3
                sim_params:
                    time_bias_error: 1.0e-3
                    time_skew_error: 1.0e-9
                    time_error: 1.0e-6
                    pos_error: [1.0e-1, 1.0e-1, 1.0e-1]
                    ang_error: [1.0e-1, 1.0e-1, 1.0e-1]
                    acc_error: [1.0e-3, 1.0e-3, 1.0e-3]
                    omg_error: [1.0e-2, 1.0e-2, 1.0e-2]
                    acc_bias_error: [1.0e-1, 1.0e-1, 1.0e-1]
                    omg_bias_error: [1.0e-1, 1.0e-1, 1.0e-1]

        camera:
            ace:
                rate: 20.0
                topic:  "/my_camera/pylon_ros2_camera_node/image_raw"
                pos_c_in_b: [0.0, 0.0, 0.0]
                ang_c_to_b: [0.5, -0.5, 0.5, -0.5]
                variance: [0.1, 0.1, 0.1, 0.1, 0.1, 0.1]
                tracker: "orb"
                pos_stability: 1.0e-9
                ang_stability: 1.0e-9
                intrinsics:
                    F: 1.0
                    c_x: 320.0
                    c_y: 240.0
                    k_1: 0.0
                    k_2: 0.0
                    p_1: 0.0
                    p_2: 0.0
                    pixel_size: 1.0e-2
                min_feat_dist: 1.0
                sim_params:
                    time_bias_error: 1.0e-3
                    time_skew_error: 1.0e-9
                    time_error: 1.0e-6
                    pos_error: [1.0e-2, 1.0e-2, 1.0e-2]
                    ang_error: [1.0e-2, 1.0e-2, 1.0e-2]

        tracker:
            orb:
                feature_detector: 2
                descriptor_extractor: 1
                descriptor_matcher: 0
                detector_threshold: 10.0
                pixel_error: 1.0
                min_feature_distance: 1.0
                min_track_length: 0
                max_track_length: 20
                sim_params:
                    feature_count: 100
                    room_size: 10.0

        gps:
            gps_1:
                rate: 5.0
                topic: "/gps1"
                pos_a_in_b: [1.0, 0, 0]
                variance: [5.0, 5.0, 5.0]
                sim_params:
                    time_bias: 0.0
                    time_skew: 0.0
                    time_error: 1.0e-9
                    pos_a_in_b: [0.0, 0.0, 0.0]
                    gps_error:  [5.0, 5.0, 5.0]
                    pos_l_in_g: [0.0, 0.0, 0.0]
                    ang_l_to_g: [1.0, 0.0, 0.0, 0.0]
                    no_errors: false

            gps_2:
                rate: 5.0
                topic: "/gps2"
                pos_a_in_b: [-1.0, 0, 0]
                variance: [5.0, 5.0, 5.0]
                sim_params:
                    time_bias: 0.0
                    time_skew: 0.0
                    time_error: 1.0e-9
                    pos_a_in_b: [0.0, 0.0, 0.0]
                    gps_error:  [5.0, 5.0, 5.0]
                    pos_l_in_g: [0.0, 0.0, 0.0]
                    ang_l_to_g: [1.0, 0.0, 0.0, 0.0]
                    no_errors: false


